protocol/x/clob/keeper/liquidations.go

```diff
@@ -105,7 +113,20 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			}
 
 			// Return unexpected errors.
-			return err
+			return nil, err
+		}
+
+		// Skip liquidation if the ClobPair for the liquidation order is in final settlement.
+		if _, found := clobPairStatuses[liquidationOrder.GetClobPairId()]; !found {
+			clobPair := k.mustGetClobPair(ctx, liquidationOrder.GetClobPairId())
+			clobPairStatuses[liquidationOrder.GetClobPairId()] = clobPair.Status
+		}
+		if clobPairStatuses[liquidationOrder.GetClobPairId()] == types.ClobPair_STATUS_FINAL_SETTLEMENT {
+			subaccountsToDeleverage = append(subaccountsToDeleverage, subaccountToDeleverage{
+				SubaccountId: liquidationOrder.GetSubaccountId(),
+				PerpetualId:  liquidationOrder.MustGetLiquidatedPerpetualId(),
+			})
```

Comment:
we don't need to append here, right? since this is going to get picked up by `GetSubaccountsWithOpenPositionsInFinalSettlementMarkets`

Reply:
my motivation in still appending here is that this negative TNC deleveraging attempt will occur earlier if I append it here rather than wait for the other function to pick it up. It's more important that we deleverage negative TNC subaccounts than it is that we do the final settlement deleveraging for non-negative TNC subaccounts